packages/wallet-ui/src/components/ui/organism/AccountSwitchModal/VisibleAccountsList/VisibleAccountsList.view.tsx

```diff
@@ -0,0 +1,89 @@
+import { Account } from 'types';
+import {
+  AccountImageStyled,
+  AccountSwitchMenuItem,
+  MenuSection,
+  MenuItemText,
+} from '../AccountSwitchModal.style';
+import { Menu } from '@headlessui/react';
+import { Box, IconButton } from '@mui/material';
+import VisibilityOffIcon from '@mui/icons-material/VisibilityOff';
+import { theme } from 'theme/default';
+import { shortenAddress } from 'utils/utils';
+
+interface Props {
+  accounts: Account[];
+  currentAddress: string;
+  switchAccount: (chainId: string, address: string) => void;
+  hideAccount: (params: {
+    chainId: string;
+    address: string;
+    currentAddress: string;
+  }) => void;
+  chainId: string;
+}
+
+export const VisibleAccountsListView = ({
+  accounts,
+  currentAddress,
+  switchAccount,
+  hideAccount,
+  chainId,
+}: Props) => (
+  <MenuSection>
+    {accounts.map((account) => {
+      const isSelected = account.address === currentAddress;
+      return (
+        <Menu.Item key={account.address}>
+          <AccountSwitchMenuItem
+            style={{
+              backgroundColor: isSelected
+                ? theme.palette.grey.grey4
+                : 'transparent',
+              borderLeft: isSelected
+                ? `4px solid ${theme.palette.secondary.main}`
+                : 'none',
+              paddingLeft: isSelected ? 15 : 22,
+            }}
+          >
+            <AccountSwitchMenuItem
+              style={{ padding: '0px' }}
+              onClick={
+                !isSelected
+                  ? () => switchAccount(chainId, account.address)
+                  : undefined
+              }
```

Comment:
same comment a HiddenAccountList

you can pass in the function

```suggestion
              onClick={onAccountSwitchClick(account)}
```